Add `allowUnused` (#24076)
microsoft/FluidFramework: .changeset/quick-pillows-jump.md

---
"@fluidframework/tree": minor
"fluid-framework": minor

> "__section": tree

Reviewer: ```suggestion
"__section": tree
"__highlight": true
```

Seems worth highlighting in the tree notes. Your call though. I think this only changes ordering and adds an icon or something to make it stand out in the notes.

Author: No existing code is impacted by this, and the API is only alpha so even people who it applies to likely should adopt it yet. So I don't think we want to highlight it.

---

> Add `allowUnused` utility

Reviewer: ```suggestion
New allowUnused utility function
```

> Add an `allowUnused` utility function which discards its type or runtime argument.

Reviewer: ```suggestion
The new `allowUnused` utility function discards its type or runtime argument.
```

> The new `allowUnused` utility function discards its type or runtime argument.

Reviewer 2: FYI: the PR description includes a version of this text with a typo ("it" instead of "its").

> This can be used to enable use of [ValidateRecursiveSchema](https://fluidframework.com/docs/api/fluid-framework/validaterecursiveschema-typealias) in environments where TypeScript is configured to reject code with unused locals.

Reviewer: ```suggestion
When TypeScript is configured to reject code with unused locals, this function can be used so that  [ValidateRecursiveSchema](https://fluidframework.com/docs/api/fluid-framework/validaterecursiveschema-typealias) works correctly.
```

"Works correctly" in my suggestion could probably be improved.

Author: ```suggestion
When TypeScript is configured to reject code with unused locals, this function can be used to suppress that error enabling use of [ValidateRecursiveSchema](https://fluidframework.com/docs/api/fluid-framework/validaterecursiveschema-typealias) to compile.
```
